spec/unit/hanami/application/component_provider_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "hanami/application"
+
+RSpec.describe Hanami::Application, "#component_provider", :application_integration do
```

Comment:
@timriley shouldn't this test go in `spec/isolation`? There the specs are run one by one by a separated Ruby process (`bundle exec rspec spec/isolation/foo_spec.rb`).

Because we're initializing Hanami global state (e.g. `Hanami.application`), it would make the test suite brittle tests like this are sharing the same Ruby process, like unit tests do.

Reply:
@jodosha The `:application_integration` metadata takes care of this, you can see how that works in `spec/support/application_integration.rb`. It's what I've used for all the specs in `spec/new_integration`, and it's working well there so far: specs run reliably and they're much, much faster than the full isolation tests.

If you want, I could move this file into that `spec/new_integration` dir? However, this actually felt like a proper unit test to me, because it's focused on testing that one specific method; it just requires the application setup to exist first.

What do you think?

Reply:
@timriley 
> The `:application_integration` metadata takes care of this, you can see how that works in spec/support/application_integration.rb.

My only concern here is: what if a unit test requires the state to not being initialized?

---

> However, this actually felt like a proper unit test to me

I agree, but because it depends on Hanami global state, IMO the better home for it is an isolation test. Isolation tests are like unit tests, but they need to be executed in their own Ruby process because they depend or modify the global state.

Reply:
Can we compromise and just make this an “integration” test and move it alongside the others I’ve made? It seems silly to make this an isolation test and slow down our suite when we’ve already done everything that’s necessary for this test to run authentically, fast, and reliably.

Reply:
@jodosha I've moved the test into the `new_integration/` folder now along with the rest of the new-style integration specs.

(FWIW, my plans is that this will subsume/merge with the existing `integration/` folder before we're done with the 2.0 effort)

Reply:
> My only concern here is: what if a unit test requires the state to not being initialized?

The before/after hooks enabled by using the `application_integration` example metadata do not actually do anything regarding application initialization. They simply set up some machinery to ensure that any state created by an application being created/booted over the course of the example is fully cleaned up again afterwards.

It doesn't boot an app by default or anything like that. In fact, it doesn't force you to write any particular style of test at all. It's just there to make sure you _can_ freely create/boot an app if that's what's necessary for your test to run.

Like I said, it works really nicely, and I've used those hooks to manage quite a range of tests now, so I don't think we need to resort to isolation testing at this point. In fact, my goal would be to reduce those to a minimum, since they run much more slowly and don't fit well with the rest of the test suite.

Does this allay your concerns?